cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/KafkaConnectService.java

```diff
@@ -56,101 +52,114 @@ public KafkaConnectService(ClusterApiUtils clusterApiUtils) {
   }
 
   public ApiResponse deleteConnector(ClusterConnectorRequest clusterConnectorRequest) {
-    log.info("Into deleteConnector {}", clusterConnectorRequest);
-    String suffixUrl =
-        clusterConnectorRequest.getEnv()
-            + "/connectors/"
-            + clusterConnectorRequest.getConnectorName();
-    Pair<String, RestTemplate> reqDetails =
-        clusterApiUtils.getRequestDetails(suffixUrl, clusterConnectorRequest.getProtocol());
-    HttpHeaders headers =
-        clusterApiUtils.createHeaders(
-            clusterConnectorRequest.getClusterIdentification(), KafkaClustersType.KAFKA_CONNECT);
-    HttpEntity<Object> request = new HttpEntity<>(headers);
-
-    try {
-      reqDetails
-          .getRight()
-          .exchange(
-              reqDetails.getLeft(),
-              HttpMethod.DELETE,
-              request,
-              new ParameterizedTypeReference<>() {});
-    } catch (HttpServerErrorException | HttpClientErrorException e) {
-      log.error("Error in deleting connector ", e);
-      return buildErrorResponseFromRestException(e, CLUSTER_API_ERR_3);
-    } catch (RestClientException ex) {
-      log.error("Error in deleting connector ", ex);
-      return ApiResponse.notOk(CLUSTER_API_ERR_3);
+    Set errMsgResponse = new HashSet();
+    for (String envUrl : getEnvironment(clusterConnectorRequest.getEnv())) {
```

Comment:
When there are multiple connect clusters (which means distributed mode), we should only check with first one. If that fails, go the next one... and so on. Internally each connect cluster will configure the same kafka bootstrap servers.

Reply:
We could write a utility method for all the endpoints, to check if the first connect cluster is available, then try updating connectors.

Reply:
Yes exactly, the loop will return a value if it is successful, if it returns a HTTP error it gets caught in the try-catch and loops for the next one. If all urls are exhausted it returns a NotOk status